### PR TITLE
Update configuration.md

### LINKED ISSUE #23859: Suggestion to hyperlink "App secrets" to relevant article
The same way it is done here:
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0

```1. [App secrets](xref:security/app-secrets) when the app runs in the `Development` environment.```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: b458a209-339d-e896-4e85-6d1c6a7aa1f9
* Version Independent ID: 0d5de37c-8f89-b6a5-3132-2c5214196bfd
* Content: [Configuration in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration)
* Content Source: [docs/core/extensions/configuration.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/configuration.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Hyperlink "App secrets" to relevant article.

Fixes #23859

### Internal preview

✔️ [Configuration in .NET](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/configuration?branch=pr-en-us-23862)

### CODE REVIEW

**Comment by Youssef1313** [comment]
There is a build warning: xref-not-found

Status checker didn't fail and is reporting: "Event is not a pull request or payload action is undefined." due to a recent change in the yml file.

File: docs/core/extensions/configuration.md
Comment (Youssef1313) [inline docs/core/extensions/configuration.md:34]:
```suggestion
1. [App secrets](/aspnet/core/security/app-secrets) when the app runs in the `Development` environment.
```

**Comment by gewarren** [comment]
> There is a build warning: xref-not-found
> 
> Status checker didn't fail and is reporting: "Event is not a pull request or payload action is undefined." due to a recent change in the yml file.

Thanks. Yes I'm waiting on a review for the PR that fixes it: https://github.com/dotnet/samples/pull/4542

**Review by tdykstra — APPROVED** [review]
LGTM after the change suggested by @Youssef1313 is accepted.

**Comment by IEvangelist** [comment]
> > There is a build warning: xref-not-found
> > Status checker didn't fail and is reporting: "Event is not a pull request or payload action is undefined." due to a recent change in the yml file.
> 
> Thanks. Yes I'm waiting on a review for the PR that fixes it: [dotnet/samples#4542](https://github.com/dotnet/samples/pull/4542)

Hi @gewarren - I just approved dotnet/samples#4542

**Pull request merged**
